src/transceiver.c

```diff
@@ -612,20 +670,27 @@ uint8_t wait_for_trans_cmd_resp(uint8_t expected_len) {
 
 bool send_trans_cmd(uint8_t expected_len, char* fmt, ...) {
     va_list args;
+    va_start(args, fmt);
+    vsnprintf((char *) command_buf, PRINT_BUF_SIZE, fmt, args);
+    va_end(args);
+    
+    // Generate check sum
+    /**
+     * Note: the 0xFF is because we want to use the entire message
+     * Could alternatively use strlen((char *) command_buf) but would 
+     * result in slower code 
+     */
+    uint32_t check_sum = crc32(command_buf, 0xFF);
```

Comment:
I think the max length should be either `COMMAND_BUF_SIZE` or `sizeof(command_buf)`